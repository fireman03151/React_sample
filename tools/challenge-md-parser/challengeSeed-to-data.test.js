/* global describe it expect beforeEach */
const mockAST = require('./fixtures/challenge-html-ast.json');
const challengeSeedToData = require('./challengeSeed-to-data');

describe('challengeSeed-to-data plugin', () => {
  const plugin = challengeSeedToData();
  let file = { data: {} };

  beforeEach(() => {
    file = { data: {} };
  });

  it('returns a function', () => {
    expect(typeof plugin).toEqual('function');
  });

  it('adds a `files` property to `file.data`', () => {
    plugin(mockAST, file);
    expect('files' in file.data).toBe(true);
  });

  it('ensures that the `files` property is an array', () => {
    plugin(mockAST, file);
    expect(Array.isArray(file.data.files)).toBe(true);
  });

  it('adds test objects to the files array following a schema', () => {
    expect.assertions(7);
    plugin(mockAST, file);
    const {
      data: { files }
    } = file;
    const testObject = files[0];
    expect(Object.keys(testObject).length).toEqual(6);
    expect(testObject).toHaveProperty('key');
    expect(testObject).toHaveProperty('ext');
    expect(testObject).toHaveProperty('name');
    expect(testObject).toHaveProperty('contents');
    expect(testObject).toHaveProperty('head');
    expect(testObject).toHaveProperty('tail');
  });

  it('only adds strings to the `files` object type', () => {
    expect.assertions(6);
    plugin(mockAST, file);
    const {
      data: { files }
    } = file;
    const testObject = files[0];
    Object.keys(testObject)
      .map(key => testObject[key])
      .forEach(value => expect(typeof value).toEqual('string'));
  });

  it('should have an output to match the snapshot', () => {
    plugin(mockAST, file);
    expect(file.data).toMatchSnapshot();
  });
});
