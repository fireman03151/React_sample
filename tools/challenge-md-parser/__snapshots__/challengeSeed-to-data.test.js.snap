// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`challengeSeed-to-data plugin should have an output to match the snapshot 1`] = `
Object {
  "files": Array [
    Object {
      "contents": "function testFunction(arg) {
  return arg;
}

testFunction('hello');
",
      "ext": "js",
      "head": "console.log('before the test');
",
      "key": "indexjs",
      "name": "index",
      "tail": "console.info('after the test');
",
    },
  ],
}
`;
